<div style="width: 100%; clear: both;">
<div style="float: left; width: 50%;">
<img src="http://www.uoc.edu/portal/_resources/common/imatges/marca_UOC/UOC_Masterbrand.jpg", align="left">
</div>
<div style="float: right; width: 50%;">
<p style="margin: 0; padding-top: 22px; text-align:right;">Visualización de datos· PEC2</p>
<p style="margin: 0; text-align:right;">2019-1 · Máster universitario en Ciencia de datos (Data science)</p>
<p style="margin: 0; text-align:right; padding-button: 100px;">Estudios de Informática, Multimedia y Telecomunicación</p>
</div>
</div>
<div style="width:100%;">&nbsp;</div>


# PEC 2: Familiarizarse con los datos

## Introducción

En esta segunda práctica de la asignatura se procederá a estudiar el conjunto de datos que servirá como referente a las siguientes prácticas. En esta ocasión, los datos elegidos para el estudio son los datos recogidos de la página peaceagreements.org, los cuales conforman una base de datos referidas a documentos sobre acuerdos de paz desde 1990, en las que se recogerían más de 2000 documentos.

## Estudio de los datos

A continuación, se procederá a analizar el conjunto de datos obtenidos de la web anteriormente mencionada. En este apartado se estudiarán las variables de la muestra, su distibución o valores nulos, entre otros.


In [1]:
import pandas as pd
import numpy as np

#Se carga el fichero con los datos
data = pd.read_csv("pax_all_agreements_data.csv", sep = ",")

#Se inspeccionan las 5 primeras líneas
data.head(5)

Con       Contp   PP                                        PPName  \
0  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
1  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
2  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
3  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   
4  Afghanistan  Government  2.0  Afghanistan: 2000s Post-intervention process   

                  Reg  AgtId  \
0  Europe and Eurasia   1739   
1  Europe and Eurasia   1923   
2  Europe and Eurasia    864   
3  Europe and Eurasia    848   
4  Europe and Eurasia    849   

                                                 Agt         Dat  \
0  Agreement between the Islamic Republic of Afgh...  2016-09-22   
1  Agreement between the two campaign teams regar...  2014-09-21   
2  Tokyo Declaration Partnership for Self-Relianc...  2012-07-08   
3  Conclusions of the Conference on Afghanistan a...  2011-12-05   
4  Istanbul Process on Regional Security and Coop...  2011-11-02   

                     Status  Lgt  ...  TjRep TjRSym TjRMa TjNR ImUN ImOth  \
0  Multiparty signed/agreed    4  ...      0      0     0    0    0     0   
1  Multiparty signed/agreed    4  ...      0      0     0    0    1     1   
2  Multiparty signed/agreed   14  ...      0      0     0    1    0     0   
3  Multiparty signed/agreed    7  ...      0      0     0    1    1     0   
4  Multiparty signed/agreed   10  ...      0      0     0    1    0     0   

  ImRef ImPK ImE  ImSrc  
0     0    0   1      1  
1     0    0   0      1  
2     0    0   1      1  
3     0    0   0      1  
4     0    0   0      1  

[5 rows x 266 columns]

Estudiando la salida anterior, se aprecia que existen más de 260 atributos, entre los que se encuentran variables numéricas y categóricas. Debido al gran volumen de variables, no se pueden explicar todas, con lo que se invita a investigar acerca de ellas a partir del documento que se puede encontrar en el siguiente fichero  https://www.peaceagreements.org/files/PA_X_codebook_Version2_June_2019.pdf.

Entre las variables principales se pueden ver Con, lugar donde se originan los conflictos, o Contp, el tipo de conflicto.

In [2]:
#Se obtiene una descripción de las variables numéricas
data.describe(include=[np.number])

PP        AgtId          Lgt   N_characters     Loc1GWNO  \
count  1782.000000  1789.000000  1789.000000    1789.000000  1769.000000   
mean     89.603255  1106.604807     6.359978   11959.409726   528.292256   
std      47.323516   635.813541    17.493402   37567.806638   227.494776   
min       1.000000     1.000000     1.000000       0.000000    41.000000   
25%      51.000000   547.000000     1.000000    2389.000000   359.000000   
50%      94.000000  1089.000000     2.000000    4362.000000   540.000000   
75%     120.000000  1671.000000     5.000000    9443.000000   678.000000   
max     203.000000  2231.000000   323.000000  908459.000000   940.000000   

         Loc2GWNO       UcdpAgr     PamAgr       CowWar          GCh  ...  \
count  232.000000  2.970000e+02  33.000000  1111.000000  1789.000000  ...   
mean   439.159483  2.903701e+10  17.272727   760.739874     0.313583  ...   
std    197.221974  3.602522e+11  10.022702   253.699787     0.773252  ...   
min      2.000000  2.400000e+01   1.000000   139.000000     0.000000  ...   
25%    345.000000  1.107000e+03   9.000000   817.000000     0.000000  ...   
50%    371.000000  1.334000e+03  17.000000   857.000000     0.000000  ...   
75%    625.000000  1.523000e+03  26.000000   905.000000     0.000000  ...   
max    770.000000  5.171991e+12  34.000000   941.000000     3.000000  ...   

             TjRep       TjRSym        TjRMa         TjNR         ImUN  \
count  1789.000000  1789.000000  1789.000000  1789.000000  1789.000000   
mean      0.174958     0.010620     0.101174     0.295137     0.199553   
std       0.550695     0.102536     0.301643     0.603943     0.399776   
min       0.000000     0.000000     0.000000     0.000000     0.000000   
25%       0.000000     0.000000     0.000000     0.000000     0.000000   
50%       0.000000     0.000000     0.000000     0.000000     0.000000   
75%       0.000000     0.000000     0.000000     0.000000     0.000000   
max       3.000000     1.000000     1.000000     3.000000     1.000000   

             ImOth        ImRef         ImPK          ImE        ImSrc  
count  1789.000000  1789.000000  1789.000000  1789.000000  1789.000000  
mean      0.241476     0.010061     0.294019     0.389603     0.999441  
std       0.428098     0.099829     0.455728     0.487797     0.023643  
min       0.000000     0.000000     0.000000     0.000000     0.000000  
25%       0.000000     0.000000     0.000000     0.000000     1.000000  
50%       0.000000     0.000000     0.000000     0.000000     1.000000  
75%       0.000000     0.000000     1.000000     1.000000     1.000000  
max       1.000000     1.000000     1.000000     1.000000     1.000000  

[8 rows x 250 columns]

En esta ejecución se pueden apreciar los estadísticos principales de los atributos, como el número de observaciones del dataset, 1782, la media, desviación típica, mínimo, máximo y percentiles.

In [3]:
#Se estudian los valores nulos
data.isnull().sum()

Con       0
Contp     0
PP        7
PPName    7
Reg       0
         ..
ImOth     0
ImRef     0
ImPK      0
ImE       0
ImSrc     0
Length: 266, dtype: int64

De esta manera, se aprecia que existen datos que poseen valores nulos.

En cuanto al análisis profundo de los datos, se realizarán a partir de la visualización a realizar. Esta visualización se titulará 'Acuerdos de paz: Países implicados, volumen y evolución en el tiempo'.

La url que lleva a la visualización a utilizar es https://public.tableau.com/profile/roberto.alexander.cervi.o#!/vizhome/PAX_PEC2/PEC2?publish=yes

A la hora de realizar esta visualización, se han elegido los aspectos más relevantes y generales sobre la base de datos. 

Tras su revisión, se han podido detectar los siguientes aspectos:

* En cuanto a los outliers, si se observa la tabla que muestra los conflictos por tipo y país de origen, se aprecia que los países Colombia, Filipinas y Bosnia y Herzegovina sobresalen del resto en cuanto a número de conflictos, lo cual se podría considerar una desviación. Por el contrario, se han visto varios países que no tienen ningún conflicto.
* Respecto al tipo de conflictos, a priori, parece que los conflictos de tipo Goverment/territoty, referidos a disparidades entre el gobierno y el territorio.
* Por otro lado, se puede apreciar que la región con más acuerdos sería África, seguido de Europa y Eurasia.

## Conclusiones y herramientas utilizadas

Tras la realización de esta PEC, se ha podido proporcionar una visión global y genérica de los datos a utilizar en las siguientes prácticas de la asigantura Visualización de datos.

De esta forma, se ha comenzado utilizando el lenguaje de programación Python para cargar el fichero de datos y analizar los distintos aspectos referidos a ellos, para seguidamente ofrecer una visualización realizada con el software Tableau, en el cual se presentan varios gráficos que recogen las tendencias y desgloses principales de los datos.

Tras esta práctica, he podido empezar y aprender a utilizar Tableau, el cual me ha parecido una herramienta muy interesante y que me invita a conocer otras herramientas de visualización con Qlik.